File: 0509-fibonacci-number/0509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n==0)return 0;
        if (n==1) return 1;
        int s1 = fib(n-1);
        int s2 = fib(n-2);
        return s1+s2;
        
    }
};

File: 0509-fibonacci-number/0509-fibonacci-number.rs
impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n == 0 {
            return 0;
        }
        if n == 1 {
            return 1;
        }
        let mut output1 =Self::fib(n-1);
        let mut output2=Self::fib(n-2);
        let mut output3 = output1 + output2;
        return output3
    }
}